redfish-core/lib/power_supply_metrics.hpp

```diff
@@ -187,110 +189,199 @@ inline void
+                messages::internalError(asyncResp->res);
                 return;
             }
 
-            std::string serviceName;
-            std::string averagePath;
-            std::string maximumPath;
+            // Set the default value to resourceNotFound, and if we confirm
+            // finding association with chassisID and powerSupplyID is correct,
+            // the error response will be cleared.
+            messages::resourceNotFound(asyncResp->res, "PowerSupplyMetrics",
+                                       "Metrics");
+
+            const std::vector<std::string>* inputHistoryItem =
+                std::get_if<std::vector<std::string>>(&(endpoints));
 
-            for (const auto& [objectPath, connectionNames] : intfSubTree)
+            if (inputHistoryItem != nullptr)
             {
-                if (objectPath.empty())
+                // more
+                if ((*inputHistoryItem).size() == 0)
                 {
-                    BMCWEB_LOG_DEBUG << "Error getting D-Bus object!";
-                    messages::internalError(aResp->res);
+                    BMCWEB_LOG_ERROR
+                        << "Input history item association error! ";
+                    messages::internalError(asyncResp->res);
                     return;
                 }
 
-                const std::string& validPath = objectPath;
-                auto psuMatchStr = powerSupplyID + "_input_power";
-                std::string psuInputPowerStr;
-                // validPath -> {psu}_input_power
-                // 5 below comes from
-                // /org/open_power/sensors/aggregation/per_30s/
-                //   0      1         2         3         4
-                // .../{psu}_input_power/[average,maximum]
-                //           5..............6
-                if (!dbus::utility::getNthStringFromPath(validPath, 5,
-                                                         psuInputPowerStr))
-                {
-                    BMCWEB_LOG_ERROR << "Got invalid path " << validPath;
-                    messages::invalidObject(aResp->res, validPath);
-                    return;
-                }
+                // Clear resourceNotFound response
+                asyncResp->res.clear();
 
-                if (psuInputPowerStr != psuMatchStr)
+                std::vector<std::string> inputHistoryPath;
+                for (const auto& objpath : *inputHistoryItem)
                 {
-                    // not this power supply, continue to next objectPath...
-                    continue;
+                    BMCWEB_LOG_DEBUG << "objpath: " << objpath;
+                    inputHistoryPath.push_back(objpath);
                 }
 
-                BMCWEB_LOG_DEBUG << "Got validPath: " << validPath;
-                for (const auto& connection : connectionNames)
-                {
-                    serviceName = connection.first;
+                const std::string& serviceName = "not here";
```

Comment:
Shouldn't this be deleted and not passed to the callback?  It doesn't provide any info.

Reply:
Hrm... I do not recall why that was put there. I think some changes removed getting the serviceName, and I forgot to cleanup not using that variable.